### DISCUSSION TITLE
Do you think a gRPC version of the Web API is worth investigating?

### DISCUSSION THREAD

**Discussion by Sean-URS** [Mimetis/Dotmim.Sync#901] [Q&A]
Hi, I was watching a demo by Steve Sanderson (BlazeOrbital) where he used gRPC to copy a database from a server to a client. Some advantages being that it uses HTTP/2 and is Polyglot.
Do you think a gRPC version of the Dotmim.Sync Web API is worth investigating?

**Comment by Mimetis** [comment]
I don't know much about **gRPC**, to be honest.
From what I've seen, it seems you need to know "in advance" the endpoints you want to expose, and the schema of the payload.
(Once again, not sure about that)
If this is the case, as DMS is generating payload on the fly depending on the table structure, it could be quite complicated

But once again, as I'm not an expert, I guess it's worth looking at it.

If you find it's feasible, I guess we can invest a little bit of time, digging into it, and see if we can spin up a `WebClient` / `WebServer` version using **gRPC**
